packages/babel-plugin-transform-destructuring/src/index.ts

```diff
@@ -13,475 +32,6 @@ export default declare((api, options) => {
   const objectRestNoSymbols =
     api.assumption("objectRestNoSymbols") ?? options.loose;
 
-  function getExtendsHelper(file) {
-    return useBuiltIns
-      ? t.memberExpression(t.identifier("Object"), t.identifier("assign"))
-      : file.addHelper("extends");
-  }
-
-  /**
-   * Test if a VariableDeclaration's declarations contains any Patterns.
-   */
-
-  function variableDeclarationHasPattern(node) {
-    for (const declar of node.declarations) {
-      if (t.isPattern(declar.id)) {
-        return true;
-      }
-    }
-    return false;
-  }
-
-  /**
-   * Test if an ArrayPattern's elements contain any RestElements.
-   */
-
-  function hasRest(pattern) {
-    for (const elem of pattern.elements) {
-      if (t.isRestElement(elem)) {
-        return true;
-      }
-    }
-    return false;
-  }
-
-  /**
-   * Test if an ObjectPattern's elements contain any RestElements.
-   */
-
-  function hasObjectRest(pattern) {
-    for (const elem of pattern.properties) {
-      if (t.isRestElement(elem)) {
-        return true;
-      }
-    }
-    return false;
-  }
-
-  const STOP_TRAVERSAL = {};
-
-  // NOTE: This visitor is meant to be used via t.traverse
-  const arrayUnpackVisitor = (node, ancestors, state) => {
-    if (!ancestors.length) {
-      // Top-level node: this is the array literal.
-      return;
-    }
-
-    if (
-      t.isIdentifier(node) &&
-      t.isReferenced(node, ancestors[ancestors.length - 1]) &&
```

Comment:
`t.isReferenced(node, ancestors[ancestors.length - 1])` is changed to `t.isReferenced(node, ancestors[ancestors.length - 1]).node`. This is a bug caught by type checker.